edgexfoundry/edgex-compose
# feat: Added experimental nanomq option to use NanoMQ as the MQTT Broker

closes #389







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have opened a PR for the related docs change (if not, why?) **README updated**
  <link to docs PR>


## Testing Instructions
From Compose Builder run the following commands
- make gen ds-virtual mqtt-bus nanomq
  - Verify ` nanomq option is not valid as MessageBus broker in secure mode` is displayed
- make run ds-virtual mqtt-bus nanomq no-secty
  -  Verify service all start and run as expected and data is flowing to App Rules engine
- make gen ds-virtual mqtt-broker nanomq asc-mqtt
   - Edit compose to replace `MQTT_BROKER_ADDRESS_PLACE_HOLDER` with `tcp://edgex-mqtt-broker:1883`
   - run `make up`
   - verify data is flowing thru `app-mqtt-export` to NanoMQ on `edgex-events` topic
